src/algorithms/time_evolution/simpleupdate.jl

```diff
@@ -98,73 +102,113 @@ function _su_ybond!(
     normalize!(A, Inf)
     normalize!(B, Inf)
     normalize!(s, Inf)
-    peps.A[row, col], peps.A[rm1, col] = A, B
+    if state isa InfinitePEPS
+        state.A[row, col], state.A[rm1, col] = A, B
+    else
+        state.A[row, col, 1], state.A[rm1, col, 1] = A, B
+    end
     env.data[2, row, col] = s
     return ϵ
 end
 
 """
-    su_iter(peps::InfinitePEPS, gate::LocalOperator, alg::SimpleUpdate, env::SUWeight; bipartite::Bool=false)
+    su_iter(peps::P, gate::LocalOperator, alg::SimpleUpdate, env::SUWeight; kwargs...) where {P <: InfiniteState}
```

Comment:
Given that `InfiniteState` is a non-exported constant, I'm not sure it helps to put this in the docstring like this. I think I would probably do something like:

```julia
"""
    su_iter(state::InfinitePEPS, gate::LocalOperator, alg::SimpleUpdate, env::SUWeight; bipartite=false)
    su_iter(densitymatrix::InfinitePEPO, gate::LocalOperator, alg::SimpleUpdate, env::SUWeight; biparatite=false, gate_side::Symbol = :both)
...
"""
```